orbit/pkg/profiles/profiles_darwin.go

```diff
@@ -104,3 +106,121 @@ var getMDMInfoFromProfilesCmd = func() ([]byte, error) {
+	}
+
+	// If an enrollment profile is assigned, the output of the command is in the form:
+	//
+	// ```
+	// Device Enrollment configuration:
+	// {
+	//     AllowPairing = 1;
+	//     AutoAdvanceSetup = 0;
+	//     AwaitDeviceConfigured = 0;
+	//     ConfigurationURL = "https://test.example.com/mdm/apple/enroll?token=1234";
+	//     ConfigurationWebURL = "https://test.example.com/mdm/apple/enroll?token=1234";
+	//     ...
+	// }
+	// ```
+	//
+	// If the host is not enrolled into an MDM, the output of the command is in the form:
+	//
+	// ```
+	// Device Enrollment configuration:
+	// (null)
+	// ```
+	//
+	// We will check that the output is at least 2 lines and contains the expected URL
+
+	lines := bytes.Split(bytes.TrimSpace(out), []byte("\n"))
+	if len(lines) < 2 {
+		return fmt.Errorf("parsing profiles output: expected at least 2 lines but got %d", len(lines))
+	}
+	if !bytes.Equal(lines[0], []byte("Device Enrollment configuration:")) {
+		return errors.New("parsing profiles output: does not match expected device enrollment configuration format")
+	}
+	if bytes.Equal(lines[1], []byte("(null)")) {
+		return errors.New("parsing profiles output: received null device enrollment configuration")
+	}
+
+	var assignedURL string
+	for _, line := range lines {
+		// Note the output may contain both ConfigurationURL and ConfigurationWebURL but we check only
+		// the latter for backwards compatibility.
+		// See https://github.com/fleetdm/fleet/blob/963b2438537de14e7e16f1f18857ed8a66d51bfc/server/mdm/apple/apple_mdm.go#L195
+		v, ok := parseEnrollmentProfileValue(line, "ConfigurationWebURL")
+		if ok {
+			assignedURL = v
+			break
+		}
+	}
+
+	if assignedURL == "" {
+		return errors.New("parsing profiles output: missing or empty configuration web url")
+	}
+
+	assigned, err := url.Parse(assignedURL)
+	if err != nil {
+		return fmt.Errorf("parsing profiles output: unable to parse configuration web url: %w", err)
+	}
+
+	if assigned.Hostname() != expected.Hostname() {
+		return fmt.Errorf(`matching configuration web url: expected '%s' but found '%s'`, expected.Hostname(), assigned.Hostname())
+	}
+
+	return nil
+}
+
+func parseEnrollmentProfileValue(line []byte, key string) (string, bool) {
+	// Output lines of `profiles show -type enrollment` take the form below:
+	// ```
+	// Device Enrollment configuration:
+	// {
+	//     AllowPairing = 1;
+	//     AutoAdvanceSetup = 0;
+	//     AwaitDeviceConfigured = 0;
+	//     ConfigurationURL = "https://test.example.com/mdm/apple/enroll?token=1234";
+	//     ConfigurationWebURL = "https://test.example.com/mdm/apple/enroll?token=1234";
+	//     ...
+	// }
+
+	// We are interested in the key-value pairs, which feature the separator " = ".
+	// Note that we want to include the spaces around the equals sign to avoid further splitting
+	// values, e.g., the url value may also contain an equals sign in the query string.
+	parts := bytes.SplitN(line, []byte(" = "), 3)
```

Comment:
if we're trimming space below, can we do this? just to be extra cautious
```suggestion
	parts := bytes.SplitN(line, []byte("="), 3)
```

Reply:
I think the separator should include the spaces so that we can distinguish " = " from other places where "=" might appear. It tracks the output consistently. 

This way `ConfigurationWebURL = "https://some.example.com?token=1234"; splits into two pieces instead of three.

Reply:
🤦 totally missed your note in the code, sorry!